src/frontend/src/common/analytics/eventTypes.ts

```diff
@@ -4,11 +4,168 @@
  * This exists to ensure consistency/clarity across types and to provide
  * a reference for data analysts working on CZ GE.
  *
- * - Each event type should have a plain-English description of what it means.
+ * --- REQUIREMENTS for adding new event types ---
+ * - Each event type should have a comment with a plain-English description
+ *   of what the event means.
  * - Event types should be written in SCREAMING_SNAKE_CASE
  * - Each entry in the enum should be a duplicate key/value string enum
+ * - If an event type entails sending additional data about what happened
+ *   (most events do), there should be a matching event TypeScript type below
+ *   in this file that shows the structure of the additional event data.
+ *   => The TS type should be named according to the event type as follows:
+ *      Event of `SOME_EVENT_FOO` <--> TS type of `AnalyticsSomeEventFoo`
  */
 export enum EVENT_TYPES {
   // User has been sent to Nextstrain to view a phylo tree
   TREE_VIEW_NEXTSTRAIN = "TREE_VIEW_NEXTSTRAIN",
+
+  // User has kicked off the creation of a Nextstrain phylo tree
+  TREE_CREATION_NEXTSTRAIN = "TREE_CREATION_NEXTSTRAIN",
+
+  // User has been sent over to UShER site. UShER will now create tree and
+  // display it on their site. External, so can't tell if tree succeeds/fails.
+  // **No `additionalEventData`**. Just tracking event occurrence right now.
+  TREE_CREATION_VIEW_USHER = "TREE_CREATION_VIEW_USHER",
+
+  // User has downloaded a file of the actual phylo tree
+  TREE_DOWNLOAD_TREE_FILE = "TREE_DOWNLOAD_TREE_FILE",
+
+  // User downloaded a template with sample identifiers and metadata showing
+  // which selected. User could then add more metadata to overlay on tree view.
+  TREE_DOWNLOAD_SELECTED_SAMPLES_TEMPLATE = "TREE_DOWNLOAD_SELECTED_SAMPLES_TEMPLATE",
+
+  // User has successfully uploaded new samples
+  SAMPLES_UPLOAD_SUCCESS = "SAMPLES_UPLOAD_SUCCESS",
+
+  // User downloading data about samples to a file(s)
+  // Does not currently address success/failure, but download failures are very
+  // rare, so generally safe to not be concerned about that aspect for now.
+  SAMPLES_DOWNLOAD_FILE = "SAMPLES_DOWNLOAD_FILE",
```

Comment:
for samples download we have TSV and fasta downloads, does this event capture both of those events or just one?

Reply:
oh i see later in it captures both events